oak/server/rust/oak_runtime/src/runtime/channel.rs

```diff
@@ -123,493 +202,96 @@ impl DotIdentifier for ChannelId {
-                let mut prev_graph_node = channel_id.dot_id();
-                for msg in channel.messages.read().unwrap().iter() {
-                    msg_counter += 1;
-                    let graph_node = format!("msg{}", msg_counter);
-                    writeln!(
-                        &mut s,
-                        "    {} -> {} [style=dashed arrowhead=none]",
-                        graph_node, prev_graph_node
-                    )
-                    .unwrap();
-                    for half in &msg.channels {
-                        match self.get_direction(*half) {
-                            Ok(HandleDirection::Write) => {
-                                writeln!(&mut s, "    {} -> {}", graph_node, half.dot_id())
-                                    .unwrap();
-                            }
-                            Ok(HandleDirection::Read) => {
-                                writeln!(&mut s, "    {} -> {}", half.dot_id(), graph_node)
-                                    .unwrap();
-                            }
-                            Err(_) => {
-                                error!(
-                                    "message in channel {} has message referencing unknown {:?}",
-                                    channel_id, half
-                                );
-                            }
-                        }
-                    }
-                    prev_graph_node = graph_node;
-                }
-            }
-            writeln!(&mut s, "  }}").unwrap();
-        }
-        s
+            .insert(thread_id, Arc::downgrade(thread));
     }
 
-    // Build an HTML page describing the `ChannelMapping` structure.
-    #[cfg(feature = "oak_debug")]
-    pub fn html(&self) -> String {
-        let mut s = String::new();
-        writeln!(&mut s, "<h2>Channels</h2>").unwrap();
-        writeln!(&mut s, "<ul>").unwrap();
-        {
-            let channels = self.channels.read().unwrap();
-            for channel_id in channels.keys().sorted() {
-                let channel = channels.get(channel_id).unwrap();
-                writeln!(
-                    &mut s,
-                    r###"<li><a href="{}">channel-{}</a> => {:?}"###,
-                    channel_id.html_path(),
-                    channel_id,
-                    channel
-                )
-                .unwrap();
+    pub fn wake_waiters(&self) {
+        // Unpark (wake up) all waiting threads that still have live references. The first thread
+        // woken can immediately read the message, and others might find `messages` is empty before
+        // they are even woken. This should not be an issue (being woken does not guarantee a
+        // message is available), but it could potentially result in some particular thread always
+        // getting first chance to read the message.
+        //
+        // If a thread is woken and finds no message it will take the `waiting_threads` lock and
```

Comment:
"will" I think &ndash; this describes the behaviour of `wait_on_channels` elsewhere.

(This is a pre-existing comment that's just been moved, and it's not a very helpful one &ndash; comments that describe how some other piece of code works at the time this code was written always get out of sync.  I'm planning an update pass over all the comments at some point, as there are also several function comments that describe how some other code uses the function rather than what it does.)